Simulation_engine/reform_nbptr.py

```diff
@@ -195,15 +195,9 @@ def fonction_parts_gagnees(nb_charge_principale=1):
             # # note 3 : Pas de personne à charge
             # - invalide
 
-            n31a = quotient_familial.not31a * (no_pac & no_alt & caseP)
-            # - ancien combatant
-            n31b = quotient_familial.not31b * (no_pac & no_alt & (caseW | caseG))
-            n31 = max_(n31a, n31b)
-            # - personne seule ayant élevé des enfants
-            n32 = quotient_familial.not32 * (
-                no_pac & no_alt & ((caseE | caseK | caseL) & not_(caseN))
-            )
-            n3 = max_(n31, n32)
+            # La formulation de l'article 195 nous pousse à faire ceci pour plus de cohérence.
+            # Ce sont les cas couverts par l'alinéa 1 de l'article 195 du CGI
+            n3 = (1.5 - enf) * (no_pac & no_alt & (caseW | caseG | caseP | ((caseE | caseK | caseL) & not_(caseN))))
```

Comment:
Aider à la compréhension de la formule en conservant un découpage par type de situation ?
```suggestion
            conditions_invalidite = caseP
            conditions_combattant = caseW | caseG
            conditions_seul_enfants = (caseE | caseK | caseL) & not_(caseN)  # demie-part supplémentaire
            n3 = (1.5 - enf) * (no_pac & no_alt & (conditions_combattant | conditions_invalidite | conditions_seul_enfants))
```

Reply:
Par ailleurs, je ne retrouve pas le `1.5` tel qu'explicité ci-dessus. Voici le seul trouvé dans l'article 195 : 
```
Par dérogation aux dispositions qui précèdent, le revenu imposable des contribuables célibataires, divorcés ou veufs n'ayant pas d'enfant à leur charge, exclusive, principale ou réputée également partagée entre les parents, est divisé par 1,5 lorsque ces contribuables (...)
```

Reply:
Vu : le facteur de division du revenu imposable est en fait le nombre de parts.
`enf` ici représente le total du nombre de parts.
`1.5 - enf` filtre les situations décrites par l'article ci-dessus. 
`n3` représente l'impact des situations décrites par l'article sur le nombre de parts.